### PR TITLE
GUI: add option stretch to full screen

### PR DESCRIPTION
**Pull request by raven02**
Courtesy of Margen67 & theboy181. This option can be used along with 21:9 fov mod

SMO_21_9

gui

### CODE REVIEW

File: src/core/frontend/framebuffer_layout.cpp [hunk 1]
```diff
@@ -27,8 +27,8 @@ FramebufferLayout DefaultFrameLayout(u32 width, u32 height) {
     // so just calculate them both even if the other isn't showing.
     FramebufferLayout res{width, height};
 
-    const float emulation_aspect_ratio{static_cast<float>(ScreenUndocked::Height) /
-                                       ScreenUndocked::Width};
+    const float emulation_aspect_ratio = (Settings::values.stretch_to_full ? static_cast<float>(height) / width : static_cast<float>(ScreenUndocked::Height) / ScreenUndocked::Width);
```
Comment (lioncash) [inline src/core/frontend/framebuffer_layout.cpp]:
```suggestion
    const float emulation_aspect_ratio = Settings::values.stretch_to_full ? static_cast<float>(height) / width : static_cast<float>(ScreenUndocked::Height) / ScreenUndocked::Width;
```

**Comment by legend800** [comment]
I might swap 30 fps and stretch screen in the list so resolution and screen options are grouped together better.

File: src/common/assert.h [hunk 2]
```diff
@@ -28,22 +28,19 @@ __declspec(noinline, noreturn)
 }
 
 #define ASSERT(_a_)                                                                                \
```
Comment (FearlessTobi) [inline src/common/assert.h]: The changes in this file need to be removed.

File: src/core/frontend/framebuffer_layout.cpp [hunk 3]
```diff
@@ -27,8 +27,11 @@ FramebufferLayout DefaultFrameLayout(u32 width, u32 height) {
     // so just calculate them both even if the other isn't showing.
     FramebufferLayout res{width, height};
 
-    const float emulation_aspect_ratio{static_cast<float>(ScreenUndocked::Height) /
-                                       ScreenUndocked::Width};
+    const float emulation_aspect_ratio =
+        Settings::values.stretch_to_full
+            ? static_cast<float>(height) / width
+            : static_cast<float>(ScreenUndocked::Height) / ScreenUndocked::Width;
+
```
Comment (FearlessTobi) [inline src/core/frontend/framebuffer_layout.cpp]: Nit: Why the newline?

File: src/core/settings.cpp [hunk 4]
```diff
@@ -94,6 +94,7 @@ void LogSettings() {
     LogSetting("Renderer_UseAccurateGpuEmulation", Settings::values.use_accurate_gpu_emulation);
     LogSetting("Renderer_UseAsynchronousGpuEmulation",
                Settings::values.use_asynchronous_gpu_emulation);
+    LogSetting("Renderer_StretchToFull", Settings::values.stretch_to_full);
```
Comment (FearlessTobi) [inline src/core/settings.cpp]: I don't see the benefit of shortening the variable name here. Imho you could just go with `stretch_to_fullscreen`.

File: src/yuzu_cmd/config.cpp [hunk 5]
```diff
@@ -388,7 +388,8 @@ void Config::ReadValues() {
         sdl2_config->GetBoolean("Renderer", "use_accurate_gpu_emulation", false);
     Settings::values.use_asynchronous_gpu_emulation =
         sdl2_config->GetBoolean("Renderer", "use_asynchronous_gpu_emulation", false);
-
+    Settings::values.stretch_to_full =
```
Comment (FearlessTobi) [inline src/yuzu_cmd/config.cpp]: Move this one line up and add the newline back which you removed.

**Comment by raven02** [comment]
Agree. Probably I can elborate more for next PR and will close this one.

**Pull request closed without merging**